### PR TITLE
Polymer eigenda develop

### PR DESCRIPTION
**Pull request by dshiell**
Merge in EigenDA support

*ON HOLD until EigenDA issues are understood.

### CODE REVIEW

File: op-program/client/program.go [hunk 1]
```diff
@@ -42,27 +43,28 @@ func Main(logger log.Logger) {
...
-		bootInfo.RollupConfig,
-		bootInfo.L2ChainConfig,
-		bootInfo.L1Head,
-		bootInfo.L2OutputRoot,
-		bootInfo.L2Claim,
-		bootInfo.L2ClaimBlockNumber,
-		l1PreimageOracle,
-		l2PreimageOracle,
-	)
+	return nil
```
Comment (Inkvi) [inline op-program/client/program.go]: This change is concerning. Worth looking into whether we can keep the old logic with minimal changes.

File: ops-bedrock/Dockerfile.stateviz [hunk 2]
```diff
@@ -2,18 +2,19 @@ FROM golang:1.20.4-alpine3.16 as builder
...
 COPY ./op-service /app/op-service
 
 WORKDIR /app/op-node
-RUN go mod download -x
+# RUN go mod download -x
 
 COPY ./op-node /app/op-node
 
-RUN go build -o ./bin/stateviz ./cmd/stateviz
+RUN --mount=type=cache,target="/root/.cache/go-build" go build -o ./bin/stateviz ./cmd/stateviz
```
Comment (Inkvi) [inline ops-bedrock/Dockerfile.stateviz]: I am curious what's the purpose of those changes?

File: ops-bedrock/docker-compose.yml [hunk 3]
```diff
@@ -80,7 +80,7 @@ services:
       --p2p.listen.ip=0.0.0.0
       --p2p.listen.tcp=9003
       --p2p.listen.udp=9003
-      --p2p.scoring.peers=light
+      --p2p.scoring=light
```
Comment (Inkvi) [inline ops-bedrock/docker-compose.yml]: Is this a correct change?

File: packages/contracts-bedrock/scripts/Deployer.sol [hunk 4]
```diff
@@ -54,60 +72,142 @@ abstract contract Deployer is Script, Artifacts {
...
+
+            string memory artifactPath = string.concat(deploymentsDir, "/", deploymentName, ".json");
+
+            uint256 numDeployments = 0;
+            try vm.readFile(artifactPath) returns (string memory res) {
+                numDeployments = stdJson.readUint(string(res), "$.numDeployments");
+                vm.removeFile(artifactPath);
+            } catch { }
+            numDeployments++;
+
```
Comment (Inkvi) [inline packages/contracts-bedrock/scripts/Deployer.sol:107]: I had to [revert a commit](https://github.com/polymerdao/optimism-dev/pull/4/commits/537de6675bc6e3f2d270e278c797401e9547331e) to be able to deploy smart contracts using this branch. If the upstream updated their docs such that we can deploy OP contracts without a problem, then we shouldn't include these changes.

File: op-program/client/program.go (hunk 1 above)
Comment (dshiell) [reply op-program/client/program.go]: yeah seems like it would bork fault proofs

File: ops-bedrock/Dockerfile.stateviz (hunk 2 above)
Comment (dshiell) [reply ops-bedrock/Dockerfile.stateviz]: streamlines the build, I removed it for now

File: ops-bedrock/docker-compose.yml (hunk 3 above)
Comment (dshiell) [reply ops-bedrock/docker-compose.yml]: no it appears wrong to me: https://github.com/Layr-Labs/optimism/blob/55a63526357cf8c672e996896baf66328fcb4563/op-node/flags/p2p_flags.go#L20

**Review by Inkvi — APPROVED** [review]

File: bedrock-devnet/devnet/__init__.py [hunk 5]
```diff
@@ -217,7 +252,7 @@ def devnet_deploy(paths):
             'go', 'run', 'cmd/main.go', 'genesis', 'l2',
             '--l1-rpc', 'http://localhost:8545',
             '--deploy-config', paths.devnet_config_path,
-            '--l1-deployments', paths.addresses_json_path,
+            '--deployment-dir', paths.deployment_dir,
```
Comment (Inkvi) [inline bedrock-devnet/devnet/__init__.py]: This one seems [incorrect](https://github.com/ethereum-optimism/optimism/blob/develop/bedrock-devnet/devnet/__init__.py#L173)

Comment (dshiell) [reply bedrock-devnet/devnet/__init__.py]: `--deployment-dir` seems to be mutually exclusive with `--l1-deployments` option: https://github.com/polymerdao/optimism-dev/blob/polymer-develop/op-node/cmd/genesis/cmd.go#L41

However, upstream `--deployment-dir` has been removed!

**Pull request closed without merging**